decode.go

```diff
@@ -156,6 +156,11 @@ func (p *parser) parse() *Node {
 		return p.document()
 	case yaml_STREAM_END_EVENT:
 		// Happens when attempting to decode an empty buffer.
+		if toplevel && len(p.event.head_comment) > 0 {
```

Comment:
Either the pre-existing comment above is incorrect, or we don't need the new variable. Also, it sounds like we should have enough state available in the parser to tell whether this is a top-level value or not. Can you provide some more details about what's going on here?

Reply:
I don't know / understand the parser well enough to be sure whether the comment is correct or not. That's why I added the parameter `toplevel`, to avoid accidentally introducing a bug here. If you prefer I can remove it.